src/main/java/edu/byu/cs/controller/SubmissionController.java

```diff
@@ -320,23 +325,17 @@ public static Submission getMostRecentSubmission(String netId, Phase phase) thro
     };
 
     public static final Route approveSubmissionPost = (req, res) -> {
-        // FIXME: These are only provided as reasonable guesses, and as a starting point.
-        // This may not be the way we want to actually go with this end-point,
-        // the only reflect the data that will need to be transferred.
-        String studentNetId = req.params(":studentNetId");
-        Phase phase = Phase.valueOf(req.params(":phase"));
-        Float approvedScore = Float.valueOf(req.params(":approvedScore"));
-        Integer penaltyPct = Integer.valueOf(req.params(":penaltyPct"));
-        String approvingNetId = req.params(":approvingNetId");
-
-        // FIXME: Validate that all of the parameters were received as valid, non-empty types.
-        // Note that the `approvedScore` field can be optionally `null`.
-
-        approveSubmission(studentNetId, phase, approvingNetId, penaltyPct);
-//        // TODO: Lookup a submission by phase, netId, and headHash to pass in as `targetSubmission`
-//        approveSubmission(studentNetId, phase, approvingNetId, penaltyPct, approvedScore, null);
-
-        // FIXME: Consider returning more interesting or hepful data.
+        User adminUser = req.session().attribute("user");
+
+        ApprovalRequest request = new Gson().fromJson(req.body(), ApprovalRequest.class);
+
+        int penalty = 0;
+        if (request.penalize()) {
+            //TODO: Put somewhere better/more configurable
```

Comment:
Can we reuse the value read in to the GradingContext variable here?

Reply:
Unfortunately, no. It's just reading in a hard coded value from a local variable. We'll need to think about this.